src/libs/safe-path/src/safe_join.rs

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::io::{Error, ErrorKind, Result};
+use std::path::{Component, Path, PathBuf};
+
+fn do_scoped_resolve<R: AsRef<Path>, U: AsRef<Path>>(
+    root: R,
+    unsafe_path: U,
+) -> Result<(PathBuf, PathBuf)> {
+    let root = root.as_ref().canonicalize()?;
+    if !root.is_absolute() {
+        return Err(Error::new(
+            ErrorKind::Other,
+            format!("Invalid root path: {}", root.display()),
+        ));
+    }
+
+    let mut nlinks = 0u32;
+    let mut curr_path = unsafe_path.as_ref().to_path_buf();
+    'restart: loop {
+        let mut subpath = PathBuf::new();
+        let mut iter = curr_path.components();
+
+        'next_comp: while let Some(comp) = iter.next() {
+            match comp {
+                Component::Prefix(_) => {
+                    return Err(Error::new(
+                        ErrorKind::Other,
+                        format!("Invalid path prefix in: {}", unsafe_path.as_ref().display()),
+                    ));
+                }
+                Component::RootDir | Component::CurDir => {
+                    continue 'next_comp;
+                }
+                Component::ParentDir => {
+                    subpath.pop();
+                }
+                Component::Normal(n) => {
+                    subpath.push(n);
+                    let path = root.join(&subpath);
+                    if let Ok(v) = path.read_link() {
+                        nlinks += 1;
+                        if nlinks > 255 {
```

Comment:
Where does this value come from? I'd consider making it a `const` with an explanatory comment.

Reply:
It's from secure_join: https://github.com/cyphar/filepath-securejoin/blob/master/join.go#L51
Changed to
```
// Follow the same configuration as
// [secure_join](https://github.com/cyphar/filepath-securejoin/blob/master/join.go#L51)
const MAX_SYMLINK_DEPTH: u32 = 255;
```